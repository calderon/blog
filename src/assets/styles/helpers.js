import { css } from "styled-components"

const srOnly = css`
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  padding: 0 !important;
  margin: -1px !important;
  overflow: hidden !important;
  clip: rect(0, 0, 0, 0) !important;
  white-space: nowrap !important;
  border: 0 !important;
`

const srOnlyFocusable = css`
  &:not(:focus) {
    ${srOnly};
  }
`

const skipLink = css`
  ${srOnly};
  ${srOnlyFocusable};
`

const _breakpoints = {
  xs: 480,
  sm: 640,
  md: 768,
  lg: 840,
  xl: 1024,
}

const devices = {
  xs: `(min-width: ${_breakpoints.xs}px)`,
  sm: `(min-width: ${_breakpoints.sm}px)`,
  md: `(min-width: ${_breakpoints.md}px)`,
  lg: `(min-width: ${_breakpoints.lg}px)`,
  xl: `(min-width: ${_breakpoints.xl}px)`,
}

export { srOnly, srOnlyFocusable, skipLink, devices }
