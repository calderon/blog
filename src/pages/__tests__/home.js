import React from "react"
import { render } from "../../../__tests__/helpers/render"

import Home from "../"

const renderComponent = ({ ...props } = {}) => render(<Home {...props} />)

describe("Home", () => {
  it("renders correctly", () => {
    const home = renderComponent()
    expect(home).toMatchSnapshot()
  })
})
