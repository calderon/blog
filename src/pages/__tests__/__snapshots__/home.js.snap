// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fzokOt bowPTa page"
      >
        <header
          class="sc-AxheI jXGDfO header"
        >
          <h1
            class="sc-AxjAm sc-Axmtr etMOqb header__title heading heading--h1"
          >
            <a
              aria-current="page"
              class="sc-AxiKw LcKoT header__title__link"
              href="/"
            >
              Daniel Calderón
            </a>
          </h1>
          <nav
            class="sc-fzoLsD ypmdL header__navigation"
          >
            <a
              class="sc-AxirZ sc-AxhUy hgqkPn header__navigation__anchor header__navigation__anchor--showMenu"
              to="#headerMenu"
            >
              <span
                class="sc-AxhCb ccMvAG"
              >
                Go to navigation menu
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-bars fa-w-14 "
                data-icon="bars"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <h1
              class="sc-AxjAm dxUEld header__navigation_title heading heading--h1"
            >
              Navigation menu
            </h1>
            <ul
              class="sc-AxgMl sc-AxmLO cahYMx header__navigation__list list"
              id="headerMenu"
            >
              <li
                class="sc-fzozJi kBScuC"
              >
                <a
                  class="sc-AxiKw LcKoT header__navigation__link header__navigation__link--blog"
                  href="/blog"
                >
                  Blog
                </a>
              </li>
              <li
                class="sc-fzozJi kBScuC"
              >
                <a
                  class="sc-AxiKw LcKoT header__navigation__link header__navigation__link--about"
                  href="/about"
                >
                  About me
                </a>
              </li>
            </ul>
            <a
              class="sc-AxirZ sc-AxhUy hgqkPn header__navigation__anchor header__navigation__anchor--hideMenu"
              to="#headerMenu"
            >
              <span
                class="sc-AxhCb ccMvAG"
              >
                Go up
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-times-circle fa-w-16 "
                data-icon="times-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </nav>
        </header>
        <aside
          class="sc-fzpans sc-fzqBZW cDfKOc"
          role="banner"
        >
          <h1
            class="sc-AxjAm dxUEld heading heading--h1"
          >
            Welcome message
          </h1>
          <p>
            Hi. I’m a 
            <mark>
              front-end developer
            </mark>
             from Sevilla.
            <br />
            I write here about things I love: 
            <mark>
              Javascript
            </mark>
            ,
             
            <mark>
              React
            </mark>
             and 
            <mark>
              CSS3
            </mark>
            .
          </p>
          <p>
            <mark>
              Be welcome.
            </mark>
          </p>
        </aside>
        <main
          class="sc-fzqNJr jRqpNG main"
        >
          <section
            class="sc-fzpans kEjOsX"
            role="complementary"
          >
            <header
              class="sc-fzplWN gwLoEI"
            >
              <h1
                class="sc-AxjAm sc-fznyAO cXUMSG heading heading--h1"
              >
                Latest articles
                 
                <small>
                  in my 
                  <a
                    class="sc-AxiKw LcKoT"
                    href="/blog"
                  >
                    blog
                  </a>
                </small>
              </h1>
            </header>
            <div
              class="sc-fznKkj eiRPwi"
            >
              <div
                class="sc-fznZeY ctNIfu"
              >
                This will be a blog post
              </div>
              <div
                class="sc-fznZeY ctNIfu"
              >
                This will be a blog post
              </div>
              <div
                class="sc-fznZeY ctNIfu"
              >
                This will be a blog post
              </div>
            </div>
          </section>
          <section
            class="sc-fzpans bxwNYX"
          >
            <header
              class="sc-fzplWN gwLoEI"
            >
              <h1
                class="sc-AxjAm sc-fznyAO cXUMSG heading heading--h1"
              >
                Contact me
              </h1>
            </header>
            <div
              class="sc-fznKkj eiRPwi"
            >
              <div
                class="sc-fznZeY ctNIfu"
              >
                <p>
                  Take a sit. How can I help you?
                </p>
              </div>
              <div
                class="sc-fznZeY ctNIfu"
              >
                Your name + Email
              </div>
              <div
                class="sc-fznZeY ctNIfu"
              >
                Your message + Send your message
              </div>
            </div>
          </section>
          <section
            class="sc-fzpans fkvCEl"
            role="contentinfo"
          >
            <h1
              class="sc-AxjAm sc-fznyAO cXUMSG heading heading--h1"
            >
              I'm also in
            </h1>
            <ul
              class="sc-AxgMl iAnzxH list"
            >
              <li
                class="sc-fzoyAV fgcDZh"
              >
                <a
                  class="sc-AxirZ cSARNv"
                  href="https://github.com/calderon"
                  rel="external me"
                  title="github"
                >
                  github
                </a>
              </li>
              <li
                class="sc-fzoyAV fgcDZh"
              >
                <a
                  class="sc-AxirZ cSARNv"
                  href="https://twitter.com/dcalderon"
                  rel="external me"
                  title="twitter"
                >
                  twitter
                </a>
              </li>
              <li
                class="sc-fzoyAV fgcDZh"
              >
                <a
                  class="sc-AxirZ cSARNv"
                  href="http://linkedin.com/in/danielcalderon"
                  rel="external me"
                  title="linkedin"
                >
                  linkedin
                </a>
              </li>
              <li
                class="sc-fzoyAV fgcDZh"
              >
                <a
                  class="sc-AxirZ cSARNv"
                  href="https://stackoverflow.com/users/7442990/daniel-calder%c3%b3n-portillo"
                  rel="external me"
                  title="stack overflow"
                >
                  stack overflow
                </a>
              </li>
            </ul>
          </section>
        </main>
        <footer
          class="sc-fzpans bekeyF"
        >
          <div
            class="sc-fznKkj hCHeFQ"
          >
            <div
              class="sc-fznZeY ctNIfu"
            >
              <p>
                This blog is made with
                 
                <a
                  class="sc-AxirZ cSARNv"
                  href="https://www.gatsbyjs.org/"
                  rel="external "
                  title="gatsby"
                >
                  gatsby
                </a>
                , hosted in
                 
                <a
                  class="sc-AxirZ cSARNv"
                  href="https://github.com/calderon/blog"
                  rel="external me"
                  title="github"
                >
                  github
                </a>
                 
                and deployed using github actions.
              </p>
            </div>
            <div
              class="sc-fznZeY mdqQG"
              opacity=".6"
            >
              <p>
                the content from the blog is
                 
                <span
                  class="copyright"
                >
                  © 2020 danielcalderon.net
                </span>
              </p>
            </div>
          </div>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzokOt bowPTa page"
    >
      <header
        class="sc-AxheI jXGDfO header"
      >
        <h1
          class="sc-AxjAm sc-Axmtr etMOqb header__title heading heading--h1"
        >
          <a
            aria-current="page"
            class="sc-AxiKw LcKoT header__title__link"
            href="/"
          >
            Daniel Calderón
          </a>
        </h1>
        <nav
          class="sc-fzoLsD ypmdL header__navigation"
        >
          <a
            class="sc-AxirZ sc-AxhUy hgqkPn header__navigation__anchor header__navigation__anchor--showMenu"
            to="#headerMenu"
          >
            <span
              class="sc-AxhCb ccMvAG"
            >
              Go to navigation menu
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-bars fa-w-14 "
              data-icon="bars"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                fill="currentColor"
              />
            </svg>
          </a>
          <h1
            class="sc-AxjAm dxUEld header__navigation_title heading heading--h1"
          >
            Navigation menu
          </h1>
          <ul
            class="sc-AxgMl sc-AxmLO cahYMx header__navigation__list list"
            id="headerMenu"
          >
            <li
              class="sc-fzozJi kBScuC"
            >
              <a
                class="sc-AxiKw LcKoT header__navigation__link header__navigation__link--blog"
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li
              class="sc-fzozJi kBScuC"
            >
              <a
                class="sc-AxiKw LcKoT header__navigation__link header__navigation__link--about"
                href="/about"
              >
                About me
              </a>
            </li>
          </ul>
          <a
            class="sc-AxirZ sc-AxhUy hgqkPn header__navigation__anchor header__navigation__anchor--hideMenu"
            to="#headerMenu"
          >
            <span
              class="sc-AxhCb ccMvAG"
            >
              Go up
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-times-circle fa-w-16 "
              data-icon="times-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                fill="currentColor"
              />
            </svg>
          </a>
        </nav>
      </header>
      <aside
        class="sc-fzpans sc-fzqBZW cDfKOc"
        role="banner"
      >
        <h1
          class="sc-AxjAm dxUEld heading heading--h1"
        >
          Welcome message
        </h1>
        <p>
          Hi. I’m a 
          <mark>
            front-end developer
          </mark>
           from Sevilla.
          <br />
          I write here about things I love: 
          <mark>
            Javascript
          </mark>
          ,
           
          <mark>
            React
          </mark>
           and 
          <mark>
            CSS3
          </mark>
          .
        </p>
        <p>
          <mark>
            Be welcome.
          </mark>
        </p>
      </aside>
      <main
        class="sc-fzqNJr jRqpNG main"
      >
        <section
          class="sc-fzpans kEjOsX"
          role="complementary"
        >
          <header
            class="sc-fzplWN gwLoEI"
          >
            <h1
              class="sc-AxjAm sc-fznyAO cXUMSG heading heading--h1"
            >
              Latest articles
               
              <small>
                in my 
                <a
                  class="sc-AxiKw LcKoT"
                  href="/blog"
                >
                  blog
                </a>
              </small>
            </h1>
          </header>
          <div
            class="sc-fznKkj eiRPwi"
          >
            <div
              class="sc-fznZeY ctNIfu"
            >
              This will be a blog post
            </div>
            <div
              class="sc-fznZeY ctNIfu"
            >
              This will be a blog post
            </div>
            <div
              class="sc-fznZeY ctNIfu"
            >
              This will be a blog post
            </div>
          </div>
        </section>
        <section
          class="sc-fzpans bxwNYX"
        >
          <header
            class="sc-fzplWN gwLoEI"
          >
            <h1
              class="sc-AxjAm sc-fznyAO cXUMSG heading heading--h1"
            >
              Contact me
            </h1>
          </header>
          <div
            class="sc-fznKkj eiRPwi"
          >
            <div
              class="sc-fznZeY ctNIfu"
            >
              <p>
                Take a sit. How can I help you?
              </p>
            </div>
            <div
              class="sc-fznZeY ctNIfu"
            >
              Your name + Email
            </div>
            <div
              class="sc-fznZeY ctNIfu"
            >
              Your message + Send your message
            </div>
          </div>
        </section>
        <section
          class="sc-fzpans fkvCEl"
          role="contentinfo"
        >
          <h1
            class="sc-AxjAm sc-fznyAO cXUMSG heading heading--h1"
          >
            I'm also in
          </h1>
          <ul
            class="sc-AxgMl iAnzxH list"
          >
            <li
              class="sc-fzoyAV fgcDZh"
            >
              <a
                class="sc-AxirZ cSARNv"
                href="https://github.com/calderon"
                rel="external me"
                title="github"
              >
                github
              </a>
            </li>
            <li
              class="sc-fzoyAV fgcDZh"
            >
              <a
                class="sc-AxirZ cSARNv"
                href="https://twitter.com/dcalderon"
                rel="external me"
                title="twitter"
              >
                twitter
              </a>
            </li>
            <li
              class="sc-fzoyAV fgcDZh"
            >
              <a
                class="sc-AxirZ cSARNv"
                href="http://linkedin.com/in/danielcalderon"
                rel="external me"
                title="linkedin"
              >
                linkedin
              </a>
            </li>
            <li
              class="sc-fzoyAV fgcDZh"
            >
              <a
                class="sc-AxirZ cSARNv"
                href="https://stackoverflow.com/users/7442990/daniel-calder%c3%b3n-portillo"
                rel="external me"
                title="stack overflow"
              >
                stack overflow
              </a>
            </li>
          </ul>
        </section>
      </main>
      <footer
        class="sc-fzpans bekeyF"
      >
        <div
          class="sc-fznKkj hCHeFQ"
        >
          <div
            class="sc-fznZeY ctNIfu"
          >
            <p>
              This blog is made with
               
              <a
                class="sc-AxirZ cSARNv"
                href="https://www.gatsbyjs.org/"
                rel="external "
                title="gatsby"
              >
                gatsby
              </a>
              , hosted in
               
              <a
                class="sc-AxirZ cSARNv"
                href="https://github.com/calderon/blog"
                rel="external me"
                title="github"
              >
                github
              </a>
               
              and deployed using github actions.
            </p>
          </div>
          <div
            class="sc-fznZeY mdqQG"
            opacity=".6"
          >
            <p>
              the content from the blog is
               
              <span
                class="copyright"
              >
                © 2020 danielcalderon.net
              </span>
            </p>
          </div>
        </div>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
